### PR TITLE
Control deprecation of v1beta1 api

### PR DESCRIPTION
**Pull request by bskiba**

### CODE REVIEW

**Comment by bskiba** [comment]
@schylek

File: vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go [hunk 1]
```diff
@@ -431,7 +436,7 @@ func (feeder *clusterStateFeeder) getSelector(vpa *vpa_types.VerticalPodAutoscal
 			{conditionType: vpa_types.ConfigDeprecated, delete: true},
 		}
 	}
-	if legacySelector != nil {
+	if legacySelector != nil && *beta1APIDeprecated {
```
Comment (schylek) [inline vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go]: This code can be easily tested in cluster_feeder_test.go

Comment (bskiba) [reply vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go]: Especially when the tests reveal you have a bug :no_mouth:. Added.

**Comment by schylek** [comment]
golint failed

**Comment by bskiba** [comment]
Fixed

**Pull request merged**